fusor-ember-cli/app/mixins/deployment-route-mixin.js

```diff
@@ -21,8 +27,11 @@ export default Ember.Mixin.create({
         }
       );
     }
-  }
+  },
 
+  updateOpenstackDefaults(deployment) {
```

Comment:
Some general thoughts on this since it came up in conversation recently:

[To me, this is a textbook case for an assertion](http://stackoverflow.com/a/1081418/319499). If I call the base implementation here, I want it to fail loudly and as close to authorship time as possible since a base method call indicates developer failure (unless it's optionally overridden? Is it okay to call this in an unimplemented state?). If I forget to override this, it will fail softly and unpredictably, and that's not good when I need to come back in 6mo and I've forgotten how this works.

I'll admit, it's a unique case since it's client-side JS. My gut is to throw something, but I don't want that to get swallowed up by a handler elsewhere since that defeats the purpose; I want the error at the call site. Alternatively, maybe it needs to be handled by the application and displayed. I'm curious about your thoughts, but I disagree with a NOOP in this case.

Food for thought:
http://stackoverflow.com/questions/7477453/best-practices-for-abstract-functions-in-javascript
http://stackoverflow.com/questions/15313418/javascript-assert/15313435#15313435